# piecharts for ever detected proteins from Cys and Lys experiments

# PANTHER analysis of protein class

"""
Created on 11.23.19
for Project MAP
@author: mfpfox
"""

In [44]:
import os
import sys
import pandas as pd

sys.path.append("/Users/mariapalafox/Desktop/Toolbox")
from all_funx import *

from IPython.display import display, HTML
pd.set_option('display.max_columns', None)
pd.options.display.max_seq_items = 2000
display(HTML("<style>.container {width:90% !important;}</style>"))

---

# version 1 pie n=3963

**below function in toolbox "Pmap_pie.py"**

In [45]:
os.chdir("/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/CYS_LYS_2012_to2018UKBpositions/REFERENCE_2018_mappedCK/SUBSETS/")

In [46]:
# ukb file with be one annotated
everC = "CYS2018_everlabeled_gene_valueCounts_2915.csv"
everK = "LYS2018_everlabeled_gene_valueCounts_2623.csv"
uni = "countsfromfastaCCDS_filter_counts_all_AA.csv"
shared3953 = "UKBID_3953set.csv"

cys = read_pd(everC)
lys = read_pd(everK)
ukb = read_pd(uni)
dropUnnamed(ukb)

shared = read_pd(shared3953)

In [47]:
print("ALL 3963 uniprot IDs: ")
print("total of all lysines detected: ", lys.counts.sum())
print("total of all cysteine detected: ", cys.counts.sum())

ALL 3963 uniprot IDs: 
total of all lysines detected:  9042
total of all cysteine detected:  6234


In [49]:
# testing main()
Cls = cys.ID.tolist()
Kls = lys.ID.tolist()

ukbC = addcolumnconditional(Cls, ukb, 'ID', 'ever_C_ID')
ukbCK = addcolumnconditional(Kls, ukbC, 'ID', 'ever_K_ID')
print("shape ukbCK labeled = ", ukbCK.shape)

# only C detected proteins
Conly = ukbCK[(ukbCK['ever_C_ID'] == "True") & (ukbCK['ever_K_ID'] == "False")]
print("TOTAL protiens with only Cys detected: ")
print(Conly.shape)
print("TOTAL Cys in the proteins: ", Conly.C.sum())
print()

# only K detected proteins
Konly = ukbCK[(ukbCK['ever_C_ID'] == "False") & (ukbCK['ever_K_ID'] == "True")]
print("TOTAL protiens with only Lysine detected: ")
print(Konly.shape)
print("TOTAL Lysine in the proteins: ", Konly.K.sum())
print()

# both C K detecte proteins
bothck = ukbCK[(ukbCK['ever_C_ID'] == "True") & (ukbCK['ever_K_ID'] == "True")]
print("TOTAL protiens with both CK detected: ")
print(bothck.shape)
print("TOTAL Lysine in the proteins: ", bothck.K.sum())
print("TOTAL Cysteine in the proteins: ", bothck.C.sum())
print()

# all C IDs
Call = ukbCK[ukbCK['ever_C_ID'] == "True"]
print("TOTAL protiens with all C detected: ")
print(Call.shape)
print("TOTAL C in the proteins: ", Call.C.sum())
print()

Kall = ukbCK[ukbCK['ever_K_ID'] == "True"]
print("TOTAL protiens with all K detected: ")
print(Kall.shape)
print("TOTAL K in the proteins: ", Kall.K.sum())


checking labeled column values: 
  ever_C_ID  Count
0     False  15517
1      True   2915

checking labeled column values: 
  ever_K_ID  Count
0     False  15809
1      True   2623

shape ukbCK labeled =  (18432, 30)
TOTAL protiens with only Cys detected: 
(1340, 30)
TOTAL Cys in the proteins:  19812

TOTAL protiens with only Lysine detected: 
(1048, 30)
TOTAL Lysine in the proteins:  34910

TOTAL protiens with both CK detected: 
(1575, 30)
TOTAL Lysine in the proteins:  61917
TOTAL Cysteine in the proteins:  14944

TOTAL protiens with all C detected: 
(2915, 30)
TOTAL C in the proteins:  34756

TOTAL protiens with all K detected: 
(2623, 30)
TOTAL K in the proteins:  96827


---
---
---


# version of PIE chart with 3,953 IDs shared across all ensembl releases

In [50]:
# 3953 ID version 
fil = shared.xref.tolist()
print("length of list: ", len(fil))
print("length of set: ", len(set(fil)))
print()
lysdrop = addcolumnconditionalDrop(fil, lys, 'ID', 'set3953_ID')
cysdrop = addcolumnconditionalDrop(fil, cys, 'ID', 'set3953_ID')
print("shape lys drop  = ", lysdrop.shape)
print("shape cys drop  = ", cysdrop.shape)

length of list:  3953
length of set:  3953

  set3953_ID  Count
0       True   2618
1      False      5
dropping false
df shape post drop:  (2618, 4)
  set3953_ID  Count
0       True   2909
1      False      6
dropping false
df shape post drop:  (2909, 4)
shape lys drop  =  (2618, 4)
shape cys drop  =  (2909, 4)


In [53]:
print("ALL 3953 uniprot IDs: ")
print("total of all lysines detected: ", lysdrop.counts.sum())
print("total of all cysteines detected: ", cysdrop.counts.sum())

ALL 3953 uniprot IDs: 
total of all lysines detected:  9034
total of all cysteines detected:  6225


In [51]:
# change df name
Cls = cysdrop.ID.tolist()
Kls = lysdrop.ID.tolist()

ukbC = addcolumnconditional(Cls, ukb, 'ID', 'ever_C_ID')
ukbCK = addcolumnconditional(Kls, ukbC, 'ID', 'ever_K_ID')
print("shape ukbCK labeled = ", ukbCK.shape)

# only C detected proteins
Conly = ukbCK[(ukbCK['ever_C_ID'] == "True") & (ukbCK['ever_K_ID'] == "False")]
print("TOTAL protiens with only Cys detected: ")
print(Conly.shape)
print("TOTAL Cys in the proteins: ", Conly.C.sum())
print()

# only K detected proteins
Konly = ukbCK[(ukbCK['ever_C_ID'] == "False") & (ukbCK['ever_K_ID'] == "True")]
print("TOTAL protiens with only Lysine detected: ")
print(Konly.shape)
print("TOTAL Lysine in the proteins: ", Konly.K.sum())
print()

# both C K detecte proteins
bothck = ukbCK[(ukbCK['ever_C_ID'] == "True") & (ukbCK['ever_K_ID'] == "True")]
print("TOTAL protiens with both CK detected: ")
print(bothck.shape)
print("TOTAL Lysine in the proteins: ", bothck.K.sum())
print("TOTAL Cysteine in the proteins: ", bothck.C.sum())
print()

# all C IDs
Call = ukbCK[ukbCK['ever_C_ID'] == "True"]
print("TOTAL protiens with all C detected: ")
print(Call.shape)
print("TOTAL C in the proteins: ", Call.C.sum())
print()

Kall = ukbCK[ukbCK['ever_K_ID'] == "True"]
print("TOTAL protiens with all K detected: ")
print(Kall.shape)
print("TOTAL K in the proteins: ", Kall.K.sum())


checking labeled column values: 
  ever_C_ID  Count
0     False  15523
1      True   2909

checking labeled column values: 
  ever_K_ID  Count
0     False  15814
1      True   2618

shape ukbCK labeled =  (18432, 30)
TOTAL protiens with only Cys detected: 
(1335, 30)
TOTAL Cys in the proteins:  19768

TOTAL protiens with only Lysine detected: 
(1044, 30)
TOTAL Lysine in the proteins:  34814

TOTAL protiens with both CK detected: 
(1574, 30)
TOTAL Lysine in the proteins:  61905
TOTAL Cysteine in the proteins:  14940

TOTAL protiens with all C detected: 
(2909, 30)
TOTAL C in the proteins:  34708

TOTAL protiens with all K detected: 
(2618, 30)
TOTAL K in the proteins:  96719


---
---
---


---
---
---


# PANTHER PROTEIN CLASSIFICATION

version = PANTHER14.1 is generated from the **2018_04 release of ReferenceProteome dataset**

<http://www.pantherdb.org/chart/pantherChart.jsp?listType=1&filterLevel=1&type=5&chartType=2&save=yes&basketItems=all&zoom=1&trackingId=626614EA3D2EAB0581A6BA3EDA4F01F5>

### unable to map some IDs
#### from cys ever n = 2915
```
 Unable to map following ids:
# UNIQUE TO C EVER LIST
Q14116

# IN LYS MISMAP LIST (5)
P50135
Q15121
Q9Y3B4
P42167
P61513
```

#### from lys ever n = 2623
```
 Unable to map following ids:
# UNIQUE TO K EVER LIST
Q13765
Q99470
A6ZKI3
Q9GZP8
P60896
P42773
Q9NRS6
O60656

# IN CYS MISMAP LIST (5)
P61513
P42167
Q9Y3B4
Q15121
P50135
```

| Not mapped in C or K ever ID list | protein primary name from ukb web |
|------------------------------------|-------------------------------------|
| P61513 | 60S ribosomal protein L37a, gene(RPL37A) |
| P42167 | Lamina-associated polypeptide 2, isoforms beta/gamma, gene(TMPO) |
| Q9Y3B4 | Splicing factor 3B subunit 6, gene(SF3B6) |
| Q15121 | Astrocytic phosphoprotein PEA-15, gene(PEA15) |
| P50135 | Histamine N-methyltransferase, gene(HNMT) |
